Build Cycle error when SmallRye OpenAPI and JFR are used in the same project

> This comes up with a somewhat different stack trace from what I see in Cryostat, but it's quite similar:
> 
> ```
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.5.3:test (default-test) on project code-with-quarkus: 
> [ERROR] 
> [ERROR] See /home/work/workspace/quarkus-jfr-circular-reproducer/target/surefire-reports for the individual test results.
> [ERROR] See dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
> [ERROR] There was an error in the forked process
> [ERROR] TestEngine with ID 'junit-jupiter' encountered a critical issue during test discovery:
> [ERROR] 
> [ERROR] (1) [ERROR] ClassSelector [className = 'org.acme.GreetingResourceTest', classLoader = null] resolution failed
> [ERROR]     Source: ClassSource [className = 'org.acme.GreetingResourceTest', filePosition = null]
> [ERROR]             at org.acme.GreetingResourceTest.<no-method>(SourceFile:0)
> [ERROR]     Cause: org.junit.platform.commons.PreconditionViolationException: Could not load class with name: org.acme.GreetingResourceTest
> [ERROR] 	at org.junit.platform.engine.discovery.ClassSelector.lambda$getJavaClass$0(ClassSelector.java:98)
> [ERROR] 	at org.junit.platform.commons.function.Try$Failure.getOrThrow(Try.java:335)
> [ERROR] 	at org.junit.platform.engine.discovery.ClassSelector.getJavaClass(ClassSelector.java:97)
> [ERROR] 	at org.junit.jupiter.engine.discovery.ClassSelectorResolver.resolve(ClassSelectorResolver.java:87)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.lambda$resolve$2(EngineDiscoveryRequestResolution.java:135)
> [ERROR] 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
> [ERROR] 	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1685)
> [ERROR] 	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
> [ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
> [ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
> [ERROR] 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
> [ERROR] 	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
> [ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
> [ERROR] 	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:189)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:126)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolveCompletely(EngineDiscoveryRequestResolution.java:92)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.run(EngineDiscoveryRequestResolution.java:83)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolver.resolve(EngineDiscoveryRequestResolver.java:148)
> [ERROR] 	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.resolveSelectors(DiscoverySelectorResolver.java:63)
> [ERROR] 	at org.junit.jupiter.engine.JupiterTestEngine.discover(JupiterTestEngine.java:70)
> [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:195)
> [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverSafely(EngineDiscoveryOrchestrator.java:174)
> [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:119)
> [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:84)
> [ERROR] 	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:104)
> [ERROR] 	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:83)
> [ERROR] 	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
> [ERROR] 	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$discover$0(InterceptingLauncher.java:33)
> [ERROR] 	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
> [ERROR] 	at org.junit.platform.launcher.core.InterceptingLauncher.discover(InterceptingLauncher.java:33)
> [ERROR] 	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
> [ERROR] 	at org.apache.maven.surefire.junitplatform.LazyLauncher.discover(LazyLauncher.java:50)
> [ERROR] 	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:52)
> [ERROR] 	at org.apache.maven.surefire.api.util.DefaultScanResult.applyFilter(DefaultScanResult.java:87)
> [ERROR] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:144)
> [ERROR] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
> [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
> [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
> [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
> [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
> [ERROR] Caused by: java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Cycle detected:
> [ERROR] 		   io.quarkus.arc.deployment.BeanArchiveProcessor#build produced class io.quarkus.arc.deployment.BeanArchiveIndexBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#smallryeOpenApiIndex produced class io.quarkus.smallrye.openapi.deployment.OpenApiFilteredIndexViewBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#addAutoFilters produced class io.quarkus.smallrye.openapi.deployment.spi.AddToOpenAPIDefinitionBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#build produced class io.quarkus.deployment.builditem.FeatureBuildItem
> [ERROR] 		to io.quarkus.jfr.deployment.JfrProcessor#registerVersion produced class io.quarkus.arc.deployment.AdditionalBeanBuildItem
> [ERROR] 		to io.quarkus.arc.deployment.BeanArchiveProcessor#build
> [ERROR] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:376)
> [ERROR] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:293)
> [ERROR] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:61)
> [ERROR] 	at io.quarkus.test.junit.AppMakerHelper.getStartupAction(AppMakerHelper.java:253)
> [ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.getOrCreateRuntimeClassLoader(FacadeClassLoader.java:544)
> [ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.getQuarkusClassLoader(FacadeClassLoader.java:441)
> [ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.loadClass(FacadeClassLoader.java:330)
> [ERROR] 	at java.base/java.lang.Class.forName0(Native Method)
> [ERROR] 	at java.base/java.lang.Class.forName(Class.java:534)
> [ERROR] 	at java.base/java.lang.Class.forName(Class.java:513)
> [ERROR] 	at org.junit.platform.commons.util.ReflectionUtils.lambda$tryToLoadClass$10(ReflectionUtils.java:886)
> [ERROR] 	at org.junit.platform.commons.function.Try.lambda$call$0(Try.java:57)
> [ERROR] 	at org.junit.platform.commons.function.Try.of(Try.java:93)
> [ERROR] 	at org.junit.platform.commons.function.Try.call(Try.java:57)
> [ERROR] 	at org.junit.platform.commons.util.ReflectionUtils.tryToLoadClass(ReflectionUtils.java:873)
> [ERROR] 	at org.junit.platform.commons.util.ReflectionUtils.tryToLoadClass(ReflectionUtils.java:810)
> [ERROR] 	at org.junit.platform.commons.support.ReflectionSupport.tryToLoadClass(ReflectionSupport.java:94)
> [ERROR] 	at org.junit.platform.engine.discovery.ClassSelector.getJavaClass(ClassSelector.java:95)
> [ERROR] 	... 38 more
> [ERROR] Caused by: io.quarkus.builder.ChainBuildException: Cycle detected:
> [ERROR] 		   io.quarkus.arc.deployment.BeanArchiveProcessor#build produced class io.quarkus.arc.deployment.BeanArchiveIndexBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#smallryeOpenApiIndex produced class io.quarkus.smallrye.openapi.deployment.OpenApiFilteredIndexViewBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#addAutoFilters produced class io.quarkus.smallrye.openapi.deployment.spi.AddToOpenAPIDefinitionBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#build produced class io.quarkus.deployment.builditem.FeatureBuildItem
> [ERROR] 		to io.quarkus.jfr.deployment.JfrProcessor#registerVersion produced class io.quarkus.arc.deployment.AdditionalBeanBuildItem
> [ERROR] 		to io.quarkus.arc.deployment.BeanArchiveProcessor#build
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:327)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:291)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:284)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.build(BuildChainBuilder.java:156)
> [ERROR] 	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:142)
> [ERROR] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:372)
> [ERROR] 	... 55 more
> [ERROR] 
> [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
> [ERROR] TestEngine with ID 'junit-jupiter' encountered a critical issue during test discovery:
> [ERROR] 
> [ERROR] (1) [ERROR] ClassSelector [className = 'org.acme.GreetingResourceTest', classLoader = null] resolution failed
> [ERROR]     Source: ClassSource [className = 'org.acme.GreetingResourceTest', filePosition = null]
> [ERROR]             at org.acme.GreetingResourceTest.<no-method>(SourceFile:0)
> [ERROR]     Cause: org.junit.platform.commons.PreconditionViolationException: Could not load class with name: org.acme.GreetingResourceTest
> [ERROR] 	at org.junit.platform.engine.discovery.ClassSelector.lambda$getJavaClass$0(ClassSelector.java:98)
> [ERROR] 	at org.junit.platform.commons.function.Try$Failure.getOrThrow(Try.java:335)
> [ERROR] 	at org.junit.platform.engine.discovery.ClassSelector.getJavaClass(ClassSelector.java:97)
> [ERROR] 	at org.junit.jupiter.engine.discovery.ClassSelectorResolver.resolve(ClassSelectorResolver.java:87)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.lambda$resolve$2(EngineDiscoveryRequestResolution.java:135)
> [ERROR] 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
> [ERROR] 	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1685)
> [ERROR] 	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
> [ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
> [ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
> [ERROR] 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
> [ERROR] 	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
> [ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
> [ERROR] 	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:189)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:126)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolveCompletely(EngineDiscoveryRequestResolution.java:92)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.run(EngineDiscoveryRequestResolution.java:83)
> [ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolver.resolve(EngineDiscoveryRequestResolver.java:148)
> [ERROR] 	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.resolveSelectors(DiscoverySelectorResolver.java:63)
> [ERROR] 	at org.junit.jupiter.engine.JupiterTestEngine.discover(JupiterTestEngine.java:70)
> [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:195)
> [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverSafely(EngineDiscoveryOrchestrator.java:174)
> [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:119)
> [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:84)
> [ERROR] 	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:104)
> [ERROR] 	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:83)
> [ERROR] 	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
> [ERROR] 	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$discover$0(InterceptingLauncher.java:33)
> [ERROR] 	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
> [ERROR] 	at org.junit.platform.launcher.core.InterceptingLauncher.discover(InterceptingLauncher.java:33)
> [ERROR] 	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
> [ERROR] 	at org.apache.maven.surefire.junitplatform.LazyLauncher.discover(LazyLauncher.java:50)
> [ERROR] 	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:52)
> [ERROR] 	at org.apache.maven.surefire.api.util.DefaultScanResult.applyFilter(DefaultScanResult.java:87)
> [ERROR] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:144)
> [ERROR] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
> [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
> [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
> [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
> [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
> [ERROR] Caused by: java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Cycle detected:
> [ERROR] 		   io.quarkus.arc.deployment.BeanArchiveProcessor#build produced class io.quarkus.arc.deployment.BeanArchiveIndexBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#smallryeOpenApiIndex produced class io.quarkus.smallrye.openapi.deployment.OpenApiFilteredIndexViewBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#addAutoFilters produced class io.quarkus.smallrye.openapi.deployment.spi.AddToOpenAPIDefinitionBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#build produced class io.quarkus.deployment.builditem.FeatureBuildItem
> [ERROR] 		to io.quarkus.jfr.deployment.JfrProcessor#registerVersion produced class io.quarkus.arc.deployment.AdditionalBeanBuildItem
> [ERROR] 		to io.quarkus.arc.deployment.BeanArchiveProcessor#build
> [ERROR] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:376)
> [ERROR] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:293)
> [ERROR] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:61)
> [ERROR] 	at io.quarkus.test.junit.AppMakerHelper.getStartupAction(AppMakerHelper.java:253)
> [ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.getOrCreateRuntimeClassLoader(FacadeClassLoader.java:544)
> [ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.getQuarkusClassLoader(FacadeClassLoader.java:441)
> [ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.loadClass(FacadeClassLoader.java:330)
> [ERROR] 	at java.base/java.lang.Class.forName0(Native Method)
> [ERROR] 	at java.base/java.lang.Class.forName(Class.java:534)
> [ERROR] 	at java.base/java.lang.Class.forName(Class.java:513)
> [ERROR] 	at org.junit.platform.commons.util.ReflectionUtils.lambda$tryToLoadClass$10(ReflectionUtils.java:886)
> [ERROR] 	at org.junit.platform.commons.function.Try.lambda$call$0(Try.java:57)
> [ERROR] 	at org.junit.platform.commons.function.Try.of(Try.java:93)
> [ERROR] 	at org.junit.platform.commons.function.Try.call(Try.java:57)
> [ERROR] 	at org.junit.platform.commons.util.ReflectionUtils.tryToLoadClass(ReflectionUtils.java:873)
> [ERROR] 	at org.junit.platform.commons.util.ReflectionUtils.tryToLoadClass(ReflectionUtils.java:810)
> [ERROR] 	at org.junit.platform.commons.support.ReflectionSupport.tryToLoadClass(ReflectionSupport.java:94)
> [ERROR] 	at org.junit.platform.engine.discovery.ClassSelector.getJavaClass(ClassSelector.java:95)
> [ERROR] 	... 38 more
> [ERROR] Caused by: io.quarkus.builder.ChainBuildException: Cycle detected:
> [ERROR] 		   io.quarkus.arc.deployment.BeanArchiveProcessor#build produced class io.quarkus.arc.deployment.BeanArchiveIndexBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#smallryeOpenApiIndex produced class io.quarkus.smallrye.openapi.deployment.OpenApiFilteredIndexViewBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#addAutoFilters produced class io.quarkus.smallrye.openapi.deployment.spi.AddToOpenAPIDefinitionBuildItem
> [ERROR] 		to io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#build produced class io.quarkus.deployment.builditem.FeatureBuildItem
> [ERROR] 		to io.quarkus.jfr.deployment.JfrProcessor#registerVersion produced class io.quarkus.arc.deployment.AdditionalBeanBuildItem
> [ERROR] 		to io.quarkus.arc.deployment.BeanArchiveProcessor#build
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:327)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:300)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:331)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:291)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.detectCycles(BuildChainBuilder.java:284)
> [ERROR] 	at io.quarkus.builder.BuildChainBuilder.build(BuildChainBuilder.java:156)
> [ERROR] 	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:142)
> [ERROR] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:372)
> [ERROR] 	... 55 more
> [ERROR] 
> [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:628)
> [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:285)
> [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:250)
> [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1337)
> [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1135)
> [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:969)
> [ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
> [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
> [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
> [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
> [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
> [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
> [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
> [ERROR] 	at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
> [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
> [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
> [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
> [ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
> [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
> [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
> [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
> [ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
> [ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
> [ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
> [ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
> [ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
> [ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
> [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
> [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
> [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
> [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
> [ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
> [ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
> [ERROR] 	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:52)
> [ERROR] 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:161)
> [ERROR] 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:73)
> ```
> 
> For some reason it looks like for Cryostat the conflict/cycle is between quarkus-jfr and hibernate-validator, but when I add those to the reproducer there is no problem. For this reproducer, adding the smallrye-openapi extension is what breaks it. 

 _Originally posted by @andrewazores in [#49532](https://github.com/quarkusio/quarkus/issues/49532#issuecomment-3275556810)_

--- a/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
+++ b/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
@@ -162,6 +162,11 @@ public class SmallRyeOpenApiProcessor {
     private static final String MANAGEMENT_ENABLED = "quarkus.smallrye-openapi.management.enabled";
 
     @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(Feature.SMALLRYE_OPENAPI);
+    }
+
+    @BuildStep
     void contributeClassesToIndex(BuildProducer<AdditionalIndexedClassesBuildItem> additionalIndexedClasses) {
         // contribute additional JDK classes to the index, because SmallRye OpenAPI will check if some
         // app types implement Map and Collection and will go through super classes until Object is reached,
@@ -889,8 +894,7 @@ public class SmallRyeOpenApiProcessor {
     }
 
     @BuildStep
-    public void build(BuildProducer<FeatureBuildItem> feature,
-            BuildProducer<GeneratedResourceBuildItem> resourceBuildItemBuildProducer,
+    public void build(BuildProducer<GeneratedResourceBuildItem> resourceBuildItemBuildProducer,
             BuildProducer<NativeImageResourceBuildItem> nativeImageResources,
             BuildProducer<OpenApiDocumentBuildItem> openApiDocumentProducer,
             OpenApiFilteredIndexViewBuildItem openApiFilteredIndexViewBuildItem,
@@ -904,8 +908,6 @@ public class SmallRyeOpenApiProcessor {
         ClassLoader loader = Thread.currentThread().getContextClassLoader();
         FilteredIndexView index = openApiFilteredIndexViewBuildItem.getIndex();
         Config config = ConfigProvider.getConfig();
-
-        feature.produce(new FeatureBuildItem(Feature.SMALLRYE_OPENAPI));
 
         List<Pattern> urlIgnorePatterns = ignoreStaticDocumentBuildItems.stream()
                 .map(IgnoreStaticDocumentBuildItem::getUrlIgnorePattern)

--- a/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
+++ b/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
@@ -326,7 +326,7 @@ public class SmallRyeOpenApiProcessor {
             // quarkus.http.cors.enabled=true
             // quarkus.http.cors.origins
             Optional<Boolean> maybeCors = c.getOptionalValue("quarkus.http.cors.enabled", Boolean.class);
-            if (!maybeCors.isPresent() || !maybeCors.get().booleanValue()) {
+            if (maybeCors.isEmpty() || !maybeCors.get()) {
                 // We need to set quarkus.http.cors.enabled=true
                 systemProperties.produce(new SystemPropertyBuildItem("quarkus.http.cors.enabled", "true"));
             }
@@ -501,30 +501,27 @@ public class SmallRyeOpenApiProcessor {
                 .map(Collection::stream)
                 .orElseGet(Stream::empty)
                 .map(securityInfo -> {
-                    switch (securityInfo.getSecurityModel()) {
-                        case jwt:
-                            return new AutoBearerTokenSecurityFilter(
-                                    config.securitySchemeName(),
-                                    config.securitySchemeDescription(),
-                                    config.getValidSecuritySchemeExtensions(),
-                                    config.jwtSecuritySchemeValue(),
-                                    config.jwtBearerFormat());
-                        case oauth2:
-                            return new AutoBearerTokenSecurityFilter(
-                                    config.securitySchemeName(),
-                                    config.securitySchemeDescription(),
-                                    config.getValidSecuritySchemeExtensions(),
-                                    config.oauth2SecuritySchemeValue(),
-                                    config.oauth2BearerFormat());
-                        case basic:
-                            return new AutoBasicSecurityFilter(
-                                    config.securitySchemeName(),
-                                    config.securitySchemeDescription(),
-                                    config.getValidSecuritySchemeExtensions(),
-                                    config.basicSecuritySchemeValue());
-                        case oidc:
+                    return switch (securityInfo.getSecurityModel()) {
+                        case jwt -> new AutoBearerTokenSecurityFilter(
+                                config.securitySchemeName(),
+                                config.securitySchemeDescription(),
+                                config.getValidSecuritySchemeExtensions(),
+                                config.jwtSecuritySchemeValue(),
+                                config.jwtBearerFormat());
+                        case oauth2 -> new AutoBearerTokenSecurityFilter(
+                                config.securitySchemeName(),
+                                config.securitySchemeDescription(),
+                                config.getValidSecuritySchemeExtensions(),
+                                config.oauth2SecuritySchemeValue(),
+                                config.oauth2BearerFormat());
+                        case basic -> new AutoBasicSecurityFilter(
+                                config.securitySchemeName(),
+                                config.securitySchemeDescription(),
+                                config.getValidSecuritySchemeExtensions(),
+                                config.basicSecuritySchemeValue());
+                        case oidc ->
                             // This needs to be a filter in runtime as the config we use to autoconfigure is in runtime
-                            return securityInfo.getOpenIDConnectInformation()
+                            securityInfo.getOpenIDConnectInformation()
                                     .map(info -> {
                                         AutoUrl openIdConnectUrl = new AutoUrl(
                                                 config.oidcOpenIdConnectUrl().orElse(null),
@@ -538,9 +535,8 @@ public class SmallRyeOpenApiProcessor {
                                                 openIdConnectUrl);
                                     })
                                     .orElse(null);
-                        default:
-                            return null;
-                    }
+                        default -> null;
+                    };
                 })
                 .filter(Objects::nonNull)
                 .findFirst();
@@ -681,7 +677,7 @@ public class SmallRyeOpenApiProcessor {
                     .stream()
                     // drop methods that specify the annotation directly
                     .filter(method -> !method.hasDeclaredAnnotation(annotation.name()))
-                    .filter(method -> isValidOpenAPIMethodForAutoAdd(method))
+                    .filter(SmallRyeOpenApiProcessor::isValidOpenAPIMethodForAutoAdd)
                     .map(method -> {
                         final ClassInfo resourceClass;
 
@@ -819,16 +815,13 @@ public class SmallRyeOpenApiProcessor {
     }
 
     private static List<MethodInfo> getMethods(ClassInfo declaringClass, IndexView index) {
-        List<MethodInfo> methods = new ArrayList<>();
-        methods.addAll(declaringClass.methods());
+        List<MethodInfo> methods = new ArrayList<>(declaringClass.methods());
 
         // Check if the method overrides a method from an interface
         for (Type interfaceType : declaringClass.interfaceTypes()) {
             ClassInfo interfaceClass = index.getClassByName(interfaceType.name());
             if (interfaceClass != null) {
-                for (MethodInfo interfaceMethod : interfaceClass.methods()) {
-                    methods.add(interfaceMethod);
-                }
+                methods.addAll(interfaceClass.methods());
             }
         }
 
@@ -919,7 +912,7 @@ public class SmallRyeOpenApiProcessor {
                 .withApplicationClassLoader(loader)
                 .withScannerClassLoader(loader)
                 .enableModelReader(true)
-                .enableStandardStaticFiles(Boolean.FALSE.equals(smallRyeOpenApiConfig.ignoreStaticDocument()))
+                .enableStandardStaticFiles(!smallRyeOpenApiConfig.ignoreStaticDocument())
                 .withResourceLocator(path -> {
                     URL locator = loader.getResource(path);
                     if (locator == null || shouldIgnore(urlIgnorePatterns, locator.toString())) {
@@ -1097,9 +1090,7 @@ public class SmallRyeOpenApiProcessor {
                 .enableAnnotationScan(false)
                 .enableStandardFilter(false);
 
-        return openApiConfig.additionalDocsDirectory()
-                .map(Collection::stream)
-                .orElseGet(Stream::empty)
+        return openApiConfig.additionalDocsDirectory().stream().flatMap(Collection::stream)
                 .map(path -> getResourceFiles(path, target))
                 .flatMap(Collection::stream)
                 .filter(path -> path.endsWith(".json") || path.endsWith(".yaml") || path.endsWith(".yml"))

